aggregator/src/aggregator.rs

```diff
@@ -3378,25 +3381,32 @@ mod tests {
         );
 
         // Recognized task ID provided
-        let response = warp::test::request()
-            .path(&format!("/hpke_config?task_id={}", task.id()))
-            .method("GET")
-            .filter(&filter)
-            .await
-            .unwrap()
-            .into_response();
+        let mut test_conn = get(&format!("/hpke_config?task_id={}", task.id()))
+            .run_async(&handler)
+            .await;
 
-        assert_eq!(response.status(), StatusCode::OK);
+        assert_eq!(test_conn.status(), Some(Status::Ok));
         assert_eq!(
-            response.headers().get(CACHE_CONTROL).unwrap(),
+            test_conn
```

Comment:
a slightly more concise alternative here would be to use [`assert_headers`](https://docs.trillium.rs/trillium_testing/macro.assert_headers.html) like 
```rust
assert_headers!(conn,
    "cache-control" => "max-age=86400",
    "content-type" => HpkeConfigList::MEDIA_TYPE)
```